### DISCUSSION TITLE
Include tags from frontmatter in the full-text search

### DISCUSSION THREAD

**Discussion by jfpedroza** [srid/emanote#397] [Ideas]
Currently, tags specified in the YAML frontmatter are not part of the content used for search. It would be nice to be able to search by those tags as well.

**Comment by srid** [comment]
Good idea, PR welcome.

**Reply by jfpedroza** [reply]
Looking at the Stork docs, I see some options.

1. Allow changing [frontmatter_handling](https://stork-search.net/docs/config-ref#frontmatter_handling) to `Ignore`, so it is included in the index.
2. Add the tags to the tile of the note.
3. Add the tags to the end of the note and use `contents` instead of `path`.

What do you think?

**Reply by srid** [reply]
I think we should do (1).

Taking a quick look, we could start by adding that config to https://github.com/EmaApps/emanote/blob/d2a64234fa327b782fcf7e49097226bea90248fe/src/Emanote/Model/Stork/Index.hs#L82-L85

**Reply by jfpedroza** [reply]
Should this be configurable from the `index.yaml`? Maybe others don't want the frontmatter to be included.

**Reply by srid** [reply]
Is there a concrete scenario (an example) where this would be considered undesirable behaviour?

**Reply by jfpedroza** [reply]
Maybe they have some keys in the metadata, like `order`, that just pollute the results, and they don't really use tags in there. I can't think of any other scenario.

**Reply by jfpedroza** [reply]
Got it working on #398 😄.

First time I have to download 12 GB to run a project 😵‍💫.

**Reply by jfpedroza** [reply]
With that, if you search `headHtml`, you get several results that simply include it in the frontmatter. But it seems Storks puts those at the end, so perhaps it is not an issue after all.
